src/armscan_env/wrapper.py

```diff
@@ -128,26 +164,116 @@ def __getattr__(self, item: str) -> Any:
         return getattr(self.env, item)
 
 
-class ArmscanEnvFactory(EnvFactory):
-    """:param name2volume: the gymnasium task/environment identifier
-    :param observation: the observation to use
-    :param reward_metric: the reward metric to use
-    :param termination_criterion: the termination criterion to use
-    :param slice_shape: the shape of the slice
-    :param max_episode_len: the maximum episode length
-    :param rotation_bounds: the bounds for the angles
-    :param translation_bounds: the bounds for the translations
-    :param render_mode_train: the render mode to use for training environments
-    :param render_mode_test: the render mode to use for test environments
-    :param render_mode_watch: the render mode to use for environments that are used to watch agent performance
-    :param venv_type: the type of vectorized environment to use
-    :param seed: the seed to use
-    :param n_stack: the number of observations to stack in a single observation
-    :param project_to_x_translation: constrains the action space to only x translation
-    :param remove_rotation_actions: removes the rotation actions from the action space
-    :param make_kwargs: additional keyword arguments to pass to the environment creation function
+class AddObservationsWrapper(Wrapper, ABC):
+    """When implementing it, make sure that additional_obs_space is available
+    before super().__init__(env) is called.
     """
 
+    def __init__(self, env: LabelmapEnv | Env, additional_obs: Observation):
+        super().__init__(env)
+        self.additional_obs = additional_obs
+        if isinstance(self.env.observation_space, Box) and isinstance(
+            self.additional_obs_space,
+            Box,
+        ):
+            self.observation_space = ConcatenatedArrayObservation.concatenate_boxes(
+                [self.env.observation_space, self.additional_obs_space],
+            )
+        else:
+            raise ValueError(
+                f"Observation spaces are not of type Box: {type(self.env.observation_space)}, {type(self.additional_obs_space)}",
+            )
+
+    @property
+    @abstractmethod
+    def additional_obs_space(self) -> gym.spaces:
+        pass
+
+    @abstractmethod
+    def get_additional_obs(
```

Comment:
rename to `get_additional_obs_array` pls to disambiguate with existing field `additional_obs`